### PR TITLE
 Upgrading to Catch2

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

Upgrading the unit-test framework to Catch2 (https://github.com/catchorg/Catch2)

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by rolanddenis** [comment]
Since I still had some comparison failures in `testRealFFT`, I have replaced the `margin` by a `scale` defined from the compared values. I hope it will work!

BTW, in the previous version of Catch, the scale was defined to `1` by default (see https://github.com/DGtal-team/DGtal/blob/master/tests/catch.hpp#L2644).

File: tests/kernel/testPointVector.cpp [hunk 1]
```diff
@@ -357,12 +357,12 @@ TEST_CASE( "4D Point Vector Unit tests" )
       REQUIRE( p1.dot(p2) == 30 );
       REQUIRE( dotProduct(p1, p2) == 30 );
 
-      REQUIRE( p1.cosineSimilarity(p1) == Approx(0.) );
-      REQUIRE( p1.cosineSimilarity(-p1) == Approx(pi) );
-      REQUIRE( p1.cosineSimilarity( Point(-2,1,-4,3) ) == Approx(pi/2) );
-      REQUIRE( cosineSimilarity(p1, p1) == Approx(0.) );
-      REQUIRE( cosineSimilarity(p1, -p1) == Approx(pi) );
-      REQUIRE( cosineSimilarity(p1, Point(-2,1,-4,3)) == Approx(pi/2) );
+      REQUIRE( p1.cosineSimilarity(p1) == Approx(0.).margin(0.00001));
```
Comment (rolanddenis) [inline tests/kernel/testPointVector.cpp]: `0.000001` instead ? (and following lines)

Comment (dcoeurjo) [reply tests/kernel/testPointVector.cpp]: Done.
Thx for the commits and comments.

**Comment by rolanddenis** [comment]
Merging, thanks!

**Comment by dcoeurjo** [comment]
🎉 ;)

**Pull request merged**